qt/CMakeLists.txt

```diff
@@ -1,5 +1,8 @@
 set_property(GLOBAL PROPERTY AUTOGEN_SOURCE_GROUP "Generated Files")
 
+# https://doc.qt.io/qt-6/macos.html#supported-versions
+set(CMAKE_OSX_DEPLOYMENT_TARGET 11)
```

Comment:
I'm late here, but:

```cmake
if(Qt_VERSION_MAJOR EQUAL 6)
    # https://doc.qt.io/qt-6/macos.html#supported-versions
    set(CMAKE_OSX_DEPLOYMENT_TARGET 11)
endif()
```

I think Qt5 is still supported in Transmission, and prev macOS versions (macOS 10.13+) are supported with Qt5.

P.S. GitHub completely messed up with my notification settings after recent outage, sorry for any delay from my side.

Reply:
Qt5 is still works in Transmission, but the Qt5 library itself is reaching EOL (or, I think it _has_ reached unpaid EOL?) so we should be migrating to Qt6

Reply:
Qt 5 has reached EOL ([source](https://endoflife.date/qt)):

![subscription timeline](https://github.com/transmission/transmission/assets/1980544/dee2428d-2342-4f33-a4ed-3388deb68762)

Qt 6.5 seems like a great target for distribution. Unfortunately, it is not in Homebrew, so it's not ideal for use in CI.

Reply:
> but the Qt5 library itself is reaching EOL

Sorry, but how is it related to my proposal?
Isn't it a separate issue?
I can create one for dropping Qt5 support, but until it's done my point stands.

Reply:
@nevack I may have misunderstood your proposal. If so, sorry about that.

I guess the question I'm wondering about is if Qt 6.x is available on all the versions of macOS that we support. If not, then we would have to decide what to do about older macOS versions.

Reply:
@nevack feel free to PR your suggestion. Although I'm not understanding who would want the unmaintained Qt5 app on an old macOS instead of the native macOS app.